reya-rust-sdk/src/http_provider.rs

```diff
@@ -1,5 +1,5 @@
 use crate::data_types;
-
+use crate::data_types::CoreProxy;
```

Comment:
hmm, think we need both the CoreProxy for the market order executions that we've already developed for the documentation 

additionally, we need to also have the OrderGatewayProxy for batch conditional order executions

Reply:
this part is not very clear, what part is executed where etc. and so how to organize it clearly. For the sdk it does not matter so much because all contract definitions are copied in one or other file anyway and not shared e.g. But we can change this later

Reply:
hmm, different functions need to be executed against different contracts with different abis / addresses

market orders we have implemented in the past before this initiative are executed against the core proxy

batch conditional orders are executed against the order gateway proxy

info about current pool price comes from the passive perp proxy